src/transforms/sample/tests.rs

```diff
@@ -67,14 +69,14 @@ fn hash_samples_at_roughly_the_configured_rate() {
             buf.into_events().next()
         })
         .count();
-    let ideal = 1.0f64 / 2.0f64;
-    let actual = total_passed as f64 / num_events as f64;
-    assert_relative_eq!(ideal, actual, epsilon = ideal * 0.5);
+    let actual = total_passed as f32 / num_events as f32;
+    assert_relative_eq!(rate.pct(), actual, epsilon = rate.pct() * 0.5);
```

Comment:
This `epsilon` value seems overly large. We allow the difference between the rate and the expected to differ by up to half of the rate? Am I misunderstanding this calculation?

Reply:
This code was unmodified by me, the diff is to hardcode the expected rate and pass it where is it needed. Maybe this was introduced to prevent flakey test failures?

Reply:
Yes, it will have been to avoid flakey tests. Technically it was modified, as `rate.pct` did not previously exist, which is then an opportunity to reexamine the scaling.